### PR TITLE
Add service for deleting old files

### PR DESCRIPTION
**Pull request by rkondratowicz**
Not scheduled/used yet.

https://tools.hmcts.net/jira/browse/BPS-578

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/sendletter/tasks/DeleteOldFilesTaskTest.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+                LocalSftpServer.SERVICE_FOLDER
+            );
+
+            DeleteOldFilesTask task = new DeleteOldFilesTask(ftp, serviceFolderMapping, Duration.ZERO);
+
+            // when
+            task.run();
+
+            //then
+            assertThat(ftp.listLetters(LocalSftpServer.SERVICE_FOLDER)).isEmpty();
```
Comment (doncem) [inline src/integrationTest/java/uk/gov/hmcts/reform/sendletter/tasks/DeleteOldFilesTaskTest.java]: Maybe worth verifying it wasn't empty before?

File: src/main/java/uk/gov/hmcts/reform/sendletter/services/ftp/FtpClient.java [hunk 2]
```diff
@@ -131,6 +131,36 @@ public void deleteReport(String reportPath) {
         });
     }
 
+    public List<FileInfo> listLetters(String serviceFolder) {
+        return runWith(sftp -> {
+            try {
+                String path = String.join("/", configProperties.getTargetFolder(), serviceFolder);
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/sendletter/services/ftp/FtpClient.java]: Do we need to worry about smth similar as in upload method:
```java
String folder = isSmokeTestFile
    ? configProperties.getSmokeTestTargetFolder()
    : String.join("/", configProperties.getTargetFolder(), serviceFolder);

String path = String.join("/", folder, file.getName());
```

File: src/main/java/uk/gov/hmcts/reform/sendletter/services/ftp/FtpClient.java [hunk 3]
```diff
@@ -131,6 +131,36 @@ public void deleteReport(String reportPath) {
...
+                return sftp
+                    .ls(path)
+                    .stream()
+                    .map(it -> new FileInfo(
+                            it.getPath(),
+                            Instant.ofEpochSecond(it.getAttributes().getMtime())
+                        )
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/sendletter/services/ftp/FtpClient.java]: Perhaps CI will pick this up - looks like incorrect indentation. I might be wrong :)

File: src/integrationTest/java/uk/gov/hmcts/reform/sendletter/tasks/DeleteOldFilesTaskTest.java (hunk 1 above)
Comment (rkondratowicz) [reply src/integrationTest/java/uk/gov/hmcts/reform/sendletter/tasks/DeleteOldFilesTaskTest.java]: done

File: src/main/java/uk/gov/hmcts/reform/sendletter/services/ftp/FtpClient.java (hunk 3 above)
Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/sendletter/services/ftp/FtpClient.java]: checkstyle / intellij think it's fine.

File: src/main/java/uk/gov/hmcts/reform/sendletter/services/ftp/FtpClient.java (hunk 2 above)
Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/sendletter/services/ftp/FtpClient.java]: smoke test letter will be handled by a separately configured job (as we will want to delete from prod as well)

**Pull request merged**